src/molecular-weight.jl

```diff
@@ -0,0 +1,260 @@
+            push!(AA_vector, AA_F_MW)
+        elseif aa_seq[i] == AA_P
+            push!(AA_vector, AA_P_MW)
+        elseif aa_seq[i] == AA_S
+            push!(AA_vector, AA_S_MW)
+        elseif aa_seq[i] == AA_T
+            push!(AA_vector, AA_T_MW)   
+        elseif aa_seq[i] == AA_W
+            push!(AA_vector, AA_W_MW)
+        elseif aa_seq[i] == AA_Y
+            push!(AA_vector, AA_Y_MW)
+        elseif aa_seq[i] == AA_V
+            push!(AA_vector, AA_V_MW)
+        elseif aa_seq[i] == AA_O
+            push!(AA_vector, AA_O_MW_MW)
+        elseif aa_seq[i] == AA_U
+            push!(AA_vector, AA_U_MW_MW)
+        elseif aa_seq[i] == AA_B
+            push!(AA_vector, AA_B_MW_MW)
+        elseif aa_seq[i] == AA_J
+            push!(AA_vector, AA_J_MW_MW)
+        elseif aa_seq[i] == AA_Z
+            push!(AA_vector, AA_Z_MW_MW)
+        elseif aa_seq[i] == AA_X
+            push!(AA_vector, AA_X_MW_MW)
+        elseif aa_seq[i] == AA_O_Gap
+            push!(AA_vector, AA_O_Gap_MW)                           
+        else
+            println("Error due to unknown amino acid at $i position")
+            return nothing
+        end
+        i = i+1
+    end
+    println("$(round(sum(AA_vector) - ((n_end - n_start - 1) * WATER_MW); digits = 3)) g/mol")
+    println("$(round(((sum(AA_vector) - ((n_end - n_start - 1) * WATER_MW))/1000); digits = 3)) kDa")
+end
+
+# Definning const of molecular weight of monophosphate in g/mol to account for the loss of water and PPi during synthesis
+const DNA_A_MW = 331.22 - WATER_MW          # DNA Adenine                 dAMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/12599
+const RNA_A_MW = 347.22 - WATER_MW          # RNA Adenine                 AMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6083
+const DNA_C_MW = 307.20 - WATER_MW          # DNA Cytosine                dCMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/13945
+const RNA_C_MW = 323.20 - WATER_MW          # RNA Cytosine                CMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6131  
+const DNA_G_MW = 347.22 - WATER_MW          # DNA Guanine                 dGMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/1135398597
+const RNA_G_MW = 363.22 - WATER_MW          # RNA Guanine                 GMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/135398631
+const DNA_T_MW = 322.21 - WATER_MW          # DNA Thymine                 dTMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/9700
+const RNA_U_MW = 324.18 - WATER_MW          # RNA Uracil                  UMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6030
+
+# Calculations following the following guidelines: https://ymc.eu/files/imported/publications/556/documents/YMC-Expert-Tip---How-to-calculate-the-MW-of-nucleic-acids.pdf
+# Defining a function, which accepts DNA sequences
+function molecular_weight(aa_seq::NucSeq{4, DNAAlphabet{4}}, unit=:"gmol", five_terminal_state=:"Hydroxyl")
```

Comment:
This should be `dna_seq`, not `aa_seq`. Otherwise, the same comments applies as to the amino acid function.